### PR TITLE
Add URL validation

### PR DESCRIPTION
**Pull request by malberts**
Refs #95

UI:
Screenshot_20220914_215411

API:
```shell
curl -X POST -H 'Content-Type: application/json' "http://localhost:8484/rest.php/wikibase-rdf/v0/mappings/Q1"   -d '[{"predicate": "owl:sameAs", "object": "http://www.w3.org/2000/01/rdf-schema#subClassOf"}, {"predicate": "rdfs:subClassOf", "object": "foo:Bar"}]'
```
Screenshot_20220914_215559

### CODE REVIEW

**Comment by malberts** [comment]
WIP beacause we need to confirm the limitation of using `filter_var( $object, FILTER_VALIDATE_URL )`.
https://www.php.net/manual/en/filter.filters.validate.php
> Validates value as URL (according to [» http://www.faqs.org/rfcs/rfc2396](http://www.faqs.org/rfcs/rfc2396)), optionally with required components. Beware a valid URL may not specify the HTTP protocol http:// so further validation may be required to determine the URL uses an expected protocol, e.g. ssh:// or mailto:. Note that the function will only find ASCII URLs to be valid; internationalized domain names (containing non-ASCII characters) will fail.

File: src/Application/Mapping.php [hunk 1]
```diff
@@ -15,6 +15,10 @@ public function __construct(
 		if ( !str_contains( $this->predicate, ':' ) ) {
 			throw new InvalidArgumentException( 'Invalid predicate' );
 		}
+
+		if ( filter_var( $object, FILTER_VALIDATE_URL ) === false ) {
```
Comment (JeroenDeDauw) [inline src/Application/Mapping.php:19]: Huh. You can access this as both a field and a local var :)

**Comment by JeroenDeDauw** [comment]
In practice this might already be good enough. Not catching some obscure stuff is already much better than letting in random junk.

**Comment by JeroenDeDauw** [comment]
But yeah, good to double-check with the stakeholders what they want exactly.

Perhaps using the same validation as Wikibase has for its URL data type makes sense. https://github.com/wikimedia/Wikibase/blob/master/repo/WikibaseRepo.datatypes.php#L342-L348

File: src/Application/Mapping.php [hunk 2]
```diff
@@ -15,6 +15,10 @@ public function __construct(
 		if ( !str_contains( $this->predicate, ':' ) ) {
 			throw new InvalidArgumentException( 'Invalid predicate' );
 		}
+
+		if ( !str_contains( $this->object, '://' ) ) {
```
Comment (malberts) [inline src/Application/Mapping.php:19]: I replaced the filter check with this simpler check. Getting the full validation in here is too difficult and it's handled by the save use case anyway.

**Comment by malberts** [comment]
This should be good enough now by using Wikibase URL validation.

**Pull request merged**